src/serialization/src/main/kotlin/org/vaccineimpact/api/serialization/Deserializer.kt

```diff
@@ -36,11 +38,22 @@ class Deserializer
             Int::class.createType() -> raw.toInt()
             Short::class.createType() -> raw.toShort()
             Float::class.createType() -> raw.toFloat()
-            Boolean::class.createType() -> raw.toBoolean()
+            Boolean::class.createType() -> raw.toBooleanValidate()
             ActivityType::class.createType() -> ActivityType.valueOf(raw.toUpperCase())
             GAVISupportLevel::class.createType() -> GAVISupportLevel.valueOf(raw.toUpperCase())
             GenderEnum::class.createType() -> GenderEnum.valueOf(raw.toUpperCase())
             else -> throw UnsupportedOperationException("org.vaccineimpact.api.serialization.Deserializer does not support target type $targetType")
         }
     }
-}
\ No newline at end of file
+
+    private fun String.toBooleanValidate(): Boolean
+    {
+        if (this.equals("true", ignoreCase = true)) {
+            return true;
+        }
+        if (this.equals("false", ignoreCase = true)) {
+            return false;
+        }
+        throw ValidationException(listOf(ErrorInfo("invalid-boolean", "Unable to parse '$this' as Boolean")))
+    }
```

Comment:
Semicolons look a bit out of place here. But could perhaps go even further...

```suggestion
    private fun String.toBooleanValidate() = when
    {
        equals("true", ignoreCase = true) -> true
        equals("false", ignoreCase = true) -> false
        else -> throw Exception("Unable to parse '$this' as Boolean")
    }
```

Reply:
Ugh, yeah, I always mindlessly pepper Kotlin with semicolons whenever I've been writing javascript. I think I'll see if I can get the linter to be more strict about them just to try and train myself out of that if nothing else. 

Ah, I did wonder about  using a `when`. Think I tend to under-use them as sometimes find them less readable than they could be, but you're right, that's much cleaner. Thanks!